packages/workflow/src/workflow.ts

```diff
@@ -380,9 +390,12 @@ function startChildWorkflowExecutionNextHandler({
         workflowIdReusePolicy: encodeWorkflowIdReusePolicy(options.workflowIdReusePolicy),
         parentClosePolicy: encodeParentClosePolicy(options.parentClosePolicy),
         cronSchedule: options.cronSchedule,
-        searchAttributes: options.searchAttributes
-          ? mapToPayloads(searchAttributePayloadConverter, options.searchAttributes)
-          : undefined,
+        searchAttributes:
+          options.searchAttributes || options.typedSearchAttributes
+            ? {
+                indexedFields: encodeUnifiedSearchAttributes(options.searchAttributes, options.typedSearchAttributes),
```

Comment:
There's no `indexFields` indirection on Workflow commands _that we emit_, because the Core's API define those fields as a map of Payload, rather than reusing the SearchAttributes message from upstream's proto (arguably, that's an error in Core's proto definitions, but I wouldn't mind changing that at this point).

Reply:
Ah, good catch.

I'm surprised that the TS compiler didn't complain about this given that with the unexpected `indexedFields` field it should be `Record<string<Record<string, Payload>>` instead of `Record<string, Payload>` like it expects.